File: src/main/java/com/rotainteligente/service/TrafficDataCollector.java
package com.rotainteligente.service;

import com.rotainteligente.datastructures.StreetStatusTable; // Corrected import
import com.rotainteligente.model.CongestionLevel;         // Corrected import
import com.rotainteligente.model.Street;                  // Corrected import

public class TrafficDataCollector {

    public static void simulateDataAndUpdateTable(StreetStatusTable table) {
        if (table == null) {
            System.err.println("StreetStatusTable cannot be null.");
            return;
        }

        // Create and add some initial streets
        Street ruaA = new Street("Rua A", CongestionLevel.LOW, false, false);
        Street ruaB = new Street("Rua B", CongestionLevel.MEDIUM, false, false);
        Street ruaC = new Street("Rua C", CongestionLevel.HIGH, true, false); // Accident on Rua C
        Street ruaD = new Street("Rua D", CongestionLevel.LOW, false, true);  // Rua D is blocked
        Street ruaE = new Street("Rua E", CongestionLevel.MEDIUM, false, false);
        Street ruaF = new Street("Rua F", CongestionLevel.LOW, false, false);

        table.addStreet(ruaA);
        table.addStreet(ruaB);
        table.addStreet(ruaC);
        table.addStreet(ruaD);
        table.addStreet(ruaE);
        table.addStreet(ruaF);

        System.out.println("Initial simulated data added to StreetStatusTable:");
        table.getAllStreets().forEach((name, street) -> System.out.println(" - " + street));

        // Simulate an update
        System.out.println("\nSimulating an update for Rua B...");
        table.updateStreetStatus("Rua B", CongestionLevel.HIGH, true, false);
        System.out.println("Rua B updated: " + table.getStreetStatus("Rua B"));

        System.out.println("\nSimulating an update for a non-existing street (Rua X)...");
        boolean updated = table.updateStreetStatus("Rua X", CongestionLevel.LOW, false, false);
        System.out.println("Update status for Rua X: " + (updated ? "Success" : "Failed (Not Found)"));
    }
    // Main method removed
}

File: app/services/traffic_data_collector.py
from app.data_structures.street_status_table import StreetStatusTable
from app.models.street import Street
from app.models.congestion_level import CongestionLevel

def simulate_data_and_update_table(table: StreetStatusTable):
    if table is None:
        print("Error: StreetStatusTable cannot be None.")
        return

    # Create and add some initial streets
    rua_a = Street("Rua A", CongestionLevel.LOW, False, False)
    rua_b = Street("Rua B", CongestionLevel.MEDIUM, False, False)
    rua_c = Street("Rua C", CongestionLevel.HIGH, True, False)  # Accident on Rua C
    rua_d = Street("Rua D", CongestionLevel.LOW, False, True)   # Rua D is blocked
    rua_e = Street("Rua E", CongestionLevel.MEDIUM, False, False)
    rua_f = Street("Rua F", CongestionLevel.LOW, False, False)

    table.add_street(rua_a)
    table.add_street(rua_b)
    table.add_street(rua_c)
    table.add_street(rua_d)
    table.add_street(rua_e)
    table.add_street(rua_f)

    print("Initial simulated data added to StreetStatusTable:")
    for name, street_obj in table.get_all_streets().items():
        print(f" - {street_obj}")

    print("\nSimulating an update for Rua B...")
    table.update_street_status("Rua B", CongestionLevel.HIGH, True, False)
    print(f"Rua B updated: {table.get_street_status('Rua B')}")

    print("\nSimulating an update for a non-existing street (Rua X)...")
    updated = table.update_street_status("Rua X", CongestionLevel.LOW, False, False)
    print(f"Update status for Rua X: {'Success' if updated else 'Failed (Not Found)'}")
